### ISSUE TITLE
[pgenlib] pip install fails with v0.81.2

### ISSUE THREAD

**Issue by aryarm** [chrchang/plink-ng#217]
## Steps to reproduce
```
conda create -y -n pgenlib-pip 'conda-forge::pip'
conda activate pgenlib-pip
pip install Pgenlib
```
Installing directly from the Github repository also fails with the same error:
```
pip install git+https://github.com/chrchang/plink-ng.git#subdirectory=2.0/Python
```
## Error log
<details><summary>Click me</summary>

> Collecting Pgenlib
>   Downloading Pgenlib-0.81.2.tar.gz (385 kB)
>      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 385.2/385.2 kB 1.7 MB/s eta 0:00:00
>   Installing build dependencies ... done
>   Getting requirements to build wheel ... done
>   Preparing metadata (pyproject.toml) ... done
> Building wheels for collected packages: Pgenlib
>   Building wheel for Pgenlib (pyproject.toml) ... error
>   error: subprocess-exited-with-error
> 
>   × Building wheel for Pgenlib (pyproject.toml) did not run successfully.
>   │ exit code: 1
>   ╰─> [12 lines of output]
>       running bdist_wheel
>       running build
>       running build_ext
>       building 'pgenlib' extension
>       creating build
>       creating build/temp.linux-x86_64-cpython-310
>       creating build/temp.linux-x86_64-cpython-310/src
>       creating build/temp.linux-x86_64-cpython-310/src/pgenlib
>       creating build/temp.linux-x86_64-cpython-310/src/plink2
>       creating build/temp.linux-x86_64-cpython-310/src/plink2/include
>       gcc -pthread -B pgenlib-pip/compiler_compat -Wno-unused-result -Wsign-compare -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem pgenlib-pip/include -fPIC -O2 -isystem pgenlib-pip/include -fPIC -Isrc/pgenlib -I/tmp/pip-build-env-w3a9ljgm/overlay/lib/python3.10/site-packages/numpy/core/include -Ipgenlib-pip/include/python3.10 -c src/pgenlib/pgenlib.cpp -o build/temp.linux-x86_64-cpython-310/src/pgenlib/pgenlib.o -std=c++98 -Wno-unused-function -Wno-macro-redefined
>       error: command 'gcc' failed: No such file or directory
>       [end of output]
> 
>   note: This error originates from a subprocess, and is likely not a problem with pip.
>   ERROR: Failed building wheel for Pgenlib
> Failed to build Pgenlib
> ERROR: Could not build wheels for Pgenlib, which is required to install pyproject.toml-based projects
</details>

## Additional info
I haven't had a chance to try to install `gcc` to try to see if that helps.
I haven't had this error when installing pgenlib before.
I also noticed that PyPI doesn't have any wheels listed for pgenlib. Uploading wheels might help.

**Comment by chrchang** [comment]
This does not fail on my Linux test machine.

While I may upload a wheel at some point, you should install gcc for now.

**Comment by aryarm** [comment]
Ok, sounds good.

Installing gcc seems to help a bit, but then I run into another error. Is there another package I should install? Here's what I'm doing
```
conda create -y -n haptools-plink -c conda-forge pip gcc
conda activate pgenlib-pip
pip install Pgenlib
```

Here's the error log:
<details><summary>Click me</summary>

> Building wheels for collected packages: Pgenlib
>   Building wheel for Pgenlib (pyproject.toml) ... error
>   error: subprocess-exited-with-error
> 
>   × Building wheel for Pgenlib (pyproject.toml) did not run successfully.
>   │ exit code: 1
>   ╰─> [14 lines of output]
>       running bdist_wheel
>       running build
>       running build_ext
>       building 'pgenlib' extension
>       creating build
>       creating build/temp.linux-x86_64-cpython-310
>       creating build/temp.linux-x86_64-cpython-310/src
>       creating build/temp.linux-x86_64-cpython-310/src/pgenlib
>       creating build/temp.linux-x86_64-cpython-310/src/plink2
>       creating build/temp.linux-x86_64-cpython-310/src/plink2/include
>       /home/amassarat/miniconda3/envs/haptools-plink/bin/x86_64-conda-linux-gnu-cc -Wno-unused-result -Wsign-compare -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /home/amassarat/miniconda3/envs/haptools-plink/include -fPIC -O2 -isystem /home/amassarat/miniconda3/envs/haptools-plink/include -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /home/amassarat/miniconda3/envs/haptools-plink/include -DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -isystem /home/amassarat/miniconda3/envs/haptools-plink/include -fPIC -Isrc/pgenlib -I/tmp/pip-build-env-ysr5ka01/overlay/lib/python3.10/site-packages/numpy/core/include -I/home/amassarat/miniconda3/envs/haptools-plink/include/python3.10 -c src/pgenlib/pgenlib.cpp -o build/temp.linux-x86_64-cpython-310/src/pgenlib/pgenlib.o -std=c++98 -Wno-unused-function -Wno-macro-redefined
>       x86_64-conda-linux-gnu-cc: fatal error: cannot execute 'cc1plus': execvp: No such file or directory
>       compilation terminated.
>       error: command '/home/amassarat/miniconda3/envs/haptools-plink/bin/x86_64-conda-linux-gnu-cc' failed with exit code 1
>       [end of output]
> 
>   note: This error originates from a subprocess, and is likely not a problem with pip.
>   ERROR: Failed building wheel

</details>

**Comment by chrchang** [comment]
On Ubuntu, the "build-essential" package should cover any other requirements.  This may be a bit different on other distributions and operating systems.

**Comment by aryarm** [comment]
Ok, installing the `build-essential` package worked!

Our team has been using `pgenlib` within a software package that we plan to publish soon. (We're very happy with it! We developed a library to read and write VCFs and PLINK2 PSAM+PVAR+PGEN files using a common python data structure, and our own benchmarks demonstrate that PLINK2 is orders of magnitude faster for the type of data we'd like to load.) So in the future, it would be helpful to have wheels, but it isn't a huge concern for us right now. In the meantime, we'll just tell users to install `build-essential` if they have any trouble.

**Comment by aryarm** [comment]
thanks for releasing v0.81.3!

is there any chance you can upload wheels for it, too? I'd be happy to help create a github action to do this automatically for you in the future, if you'd like

**Comment by chrchang** [comment]
Manylinux wheels are now available for 0.82.1.

**Comment by aryarm** [comment]
great, thank you so much!
